sklearn/utils/estimator_checks.py

```diff
@@ -760,7 +763,13 @@ def _check_transformer(name, transformer_orig, X, y):
         # raises error on malformed input for transform
         if hasattr(X, 'T'):
             # If it's not an array, it does not have a 'T' property
-            assert_raises(ValueError, transformer.transform, X.T)
+            with assert_raises(ValueError, msg="The classifer {name} does not "
+                               "raise an error when incorrect/malformed input "
+                               " for predict is passed. Number of features in "
+                               "predict dataset does not match the number of f"
+                               "eatures in fit dataset."
+                               " Perhaps use check_array"):
```

Comment:
I don't think ``check_array`` helps here.

Reply:
Would it be fine if I only replace this statement with the one you mentioned  "The transformer {} does not raise an error when the number of features in transform is different from the number of features in fit" ?

Reply:
sure (though you want to keep the name).